packages/eslint-plugin/tests/rules/prefer-includes.test.ts

```diff
@@ -89,6 +89,31 @@ ruleTester.run('prefer-includes', rule, {
         something.test(a)
       }
     `,
+    // https://github.com/typescript-eslint/typescript-eslint/issues/471
+    `
+      class MyList<T> {
+        indexOf(value: T): number {
+          return -1;
+        }
+
+        includes(value: T): boolean {
+          return this.indexOf(value) !== -1
+        }
+      }
+    `,
+    `
+    class Wrapper {
+      includes(value: T): boolean {
+        class MyList<T> {
+          indexOf(value: T): number {
+            return -1;
```

Comment:
this test case will pass regardless of your changes, as there's no call expression.